tests/cxx/typedef_in_template.cc

```diff
@@ -7,51 +7,54 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include "tests/cxx/typedef_in_template-d1.h"
-
-template<class T>
-class Container {
```

Comment:
One thing I really liked with this testcase was how it showed how something STL-container-like might behave. Granted, it was too simplified to be a real example, but it told a nice story.

It might be nice to add little breadcrumb comments to describe what real-life libstdc++ or libc++ constructs your example classes map to, because that makes it easier to to see how the testcase maps to the real world, and notice when they drift apart (e.g. if we get bug reports for libstdc++ template stuff in the future, it's nice to know where to look for related test coverage)

Reply:
I've added a comment here.  There's less direct correlation than in the previous testcase, but it might help explain what's going on.